security/pkg/server/ca/server.go

```diff
@@ -82,7 +92,6 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 	s.monitoring.CSR.Increment()
 	caller := s.authenticate(ctx)
 	if caller == nil {
-		serverCaLog.Warn("request authentication failure")
```

Comment:
Why this line of code is removed?

Reply:
we don't need to log it twice, authenticate already logs

Reply:
If caller == nil, the function will return (i.e., line 96: nil, status.Error(codes.Unauthenticated, "request authenticate failure")) and will not reach the new logging statement (i.e., line 348, serverCaLog.Warnf("Authentication failed for %v: %s", getConnectionAddress(ctx), errMsg)). 
Can you elaborate on logging it twice?

Reply:
line 348 happens prior to returning. Caller can only be nil if we have already reached l348

Reply:
Thanks for the clarification!